capellambse/model/common/accessors.py

```diff
@@ -590,8 +698,20 @@ def delete(
     ) -> None:
         assert obj._model is elmlist._model
 
-        elmlist._model._loader.idcache_remove(obj._element)
-        elmlist._parent._element.remove(obj._element)
+        with contextlib.ExitStack() as stack:
+            for ref, attr, _ in elmlist._model.find_references(obj):
+                acc = getattr(type(ref), attr)
+                if acc is self:
+                    continue
+                stack.enter_context(acc.purge_references(ref, obj))
```

Comment:
I'd like to understand what happens here and why we use a stack. I hope you can teach us in the next daily.